### PR TITLE
Gamified - Merging PBIs from the first 2 sprints

### PR DESCRIPTION
**Pull request by Mafusn**
## Description

Briefly describe the purpose and context of this pull request.

This pull request includes the implementation of the following PBIs:

- [App Login] As non-logged in user, I want to have the ability to preview the app, so that I can understand the application before register/login
- [App Home] As a waste picker, I want to view when I don't have an active course (empty state)
- [App Profile] As a waste picker, I would like to be able to see and edit my profile, in order to see what personal information is connected to my profile and edit in case something is incorrect or not up to date
- [App Profile] As a waste-picker, I want to have the ability to delete my account

## Changes

- Delete user implemented
- Profile settings page created
- Implemented ability to update first name, last name and email
- Added a welcome screen
- Updated screen when the user has no active courses
- Unit tests made
- Cleanup in all touched files
- Updated some of the routes in api.js and userApi.js

## Related Issues

Deletion of user can end up not working, as it hasn’t been tested with a logged in user. The same applies to update of first name, last name and email.

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)
Simulator_Screenshot_-_iPhone_15_Pro_-_2023-10-06_at_14 50 56
Simulator_Screenshot_-_iPhone_15_Pro_-_2023-10-06_at_14 58 28
Simulator_Screenshot_-_iPhone_15_Pro_-_2023-10-06_at_14 55 48

Profile settings page

## Notes for Reviewers

### CODE REVIEW

**Review by Havkost — CHANGES_REQUESTED** [review]
## Changes requested
Nice work, only got a few comments for the merge, keep at it.

File: eml/App.js [hunk 1]
```diff
@@ -169,29 +182,79 @@ function HomeStack() {
...
+      fetchData();
+    }, 3000);
+  }, []);
+
+  return { initialRoute, isLoading };
+}
+
 // Change InitialRouteName to HomeStack if you want to skip Login Screen
 export default function App() {
-  return isFontsLoaded() ? (
```
[the remark this answers is not in the document]
Comment (Jatewo) [reply eml/App.js:174]: > Seems to me you've removed the isFontsLoaded check, is this on purpose?

I believe this is the only place where we actually need this check, so it probably shouldn't be removed

File: eml/components/profileSettings/changeEmailButton.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+          id,
+          firstName: firstName,
+          lastName: lastName,
+          email: newEmail,
+        };
+
+        await AsyncStorage.setItem(USER_INFO, JSON.stringify(updatedProfile));
+        setEmailModalVisible(false);
+        } catch (error) {
+          Alert.alert('Alerta', 'Erro ao atualizar o e-mail, tente novamente: ', error.message);
```
[the remark this answers is not in the document]
Comment (Jatewo) [reply eml/components/profileSettings/changeEmailButton.js:72]: +1

File: eml/constants/Fonts.js [hunk 3]
```diff
@@ -3,6 +3,8 @@ import { useFonts } from 'expo-font';
 export const isFontsLoaded = () => {
   const [fontsLoaded] = useFonts({
     'Montserrat-Regular': require('../assets/fonts/Montserrat-Regular.ttf'),
```
Comment (Jatewo) [inline eml/constants/Fonts.js:5]: We should probably have all the fonts in here, and also make sure they are defined in tailwind

**Review by Jatewo — CHANGES_REQUESTED** [review]
Good work. I have some comments with changes requested and a few questions. Furthermore, in general, I'd like some comments/documentation in all files, which is also needed to adhere to our DoD. You are also missing quite a few tests as far as I can see. There seems to be several files without any tests made for them.

File: eml/App.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (benceisonline) [reply eml/App.js:174]: We do not need it in app.js, but in screens that use fonts.

File: eml/components/loading/Loading.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import { Text, Image } from "react-native";
+import { BgLinearGradient } from "../../constants/BgLinearGradient";
+import { SafeAreaView } from "react-native-safe-area-context";
+import { isFontsLoaded } from "../../constants/Fonts";
+
+const LoadingScreen = () => {
+  const logo = require("../../assets/images/logo.png");
+
+  if (!isFontsLoaded()) {
```
[the remark this answers is not in the document]
Comment (benceisonline) [reply eml/components/loading/Loading.jsx:10]: we get an error if we add isfontsloaded to app.js, and also if we remove from screens

**Pull request closed without merging**